File: cdk_integration_tests/src/python/WAFEnabled/fail__1__.py
from aws_cdk import core
from aws_cdk import aws_cloudfront as cloudfront

class CloudFrontDistributionStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Create a CloudFront distribution
        distribution = cloudfront.CfnDistribution(
            self,
            "MyCloudFrontDistribution",
            distribution_config={
                "defaultCacheBehavior": {
                    # Configure your cache behavior as needed
                },
                "enabled": True,
            }
        )

app = core.App()
CloudFrontDistributionStack(app, "CloudFrontDistributionStack")
app.synth()

File: cdk_integration_tests/src/typescript/WAFEnabled/fail__1__.ts
import { App, Stack } from 'aws-cdk-lib';
import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
import { Construct } from 'constructs';

class CloudFrontDistributionStack extends Stack {
  constructor(scope: Construct, id: string, props?: {}) {
    super(scope, id, props);

    // Create a CloudFront distribution
    new cloudfront.CfnDistribution(this, 'MyCloudFrontDistribution', {
      distributionConfig: {
        defaultCacheBehavior: {
          // Configure your cache behavior as needed
          viewerProtocolPolicy: 'allow-all', // Example configuration
          targetOriginId: 'myTargetOrigin', // Example configuration, needs to match an origin
          forwardedValues: {
            queryString: true,
            cookies: { forward: 'none' },
          },
        },
        enabled: true,
        // Other distributionConfig properties as needed
      },
    });
  }
}

const app = new App();
new CloudFrontDistributionStack(app, "CloudFrontDistributionStack");
app.synth();
